crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -104,36 +95,54 @@ impl Into<StoreError> for SledStoreError {
             SledStoreError::Json(e) => StoreError::Json(e),
             SledStoreError::Identifier(e) => StoreError::Identifier(e),
             SledStoreError::Encryption(e) => match e {
-                store_key::Error::Random(e) => StoreError::Encryption(e.to_string()),
-                store_key::Error::Serialization(e) => StoreError::Json(e),
-                store_key::Error::Encryption(e) => StoreError::Encryption(e),
+                KeyEncryptionError::Random(e) => StoreError::Encryption(e.to_string()),
+                KeyEncryptionError::Serialization(e) => StoreError::Json(e),
+                KeyEncryptionError::Encryption(e) => StoreError::Encryption(e.to_string()),
+                KeyEncryptionError::Version(found, expected) => StoreError::Encryption(format!(
+                    "Bad Database Encryption Version: expected {} found {}",
+                    expected, found
+                )),
+                KeyEncryptionError::Length(found, expected) => StoreError::Encryption(format!(
+                    "The database key an invalid length: expected {} found {}",
+                    expected, found
+                )),
             },
             SledStoreError::StoreError(e) => e,
             _ => StoreError::Backend(anyhow!(self)),
         }
     }
 }
 
-type Result<A, E = SledStoreError> = std::result::Result<A, E>;
+const SESSION: &str = "session";
+const ACCOUNT_DATA: &str = "account-data";
+const MEMBER: &str = "member";
+const PROFILE: &str = "profile";
+const ROOM_STATE: &str = "room-state";
+const PRESENCE: &str = "presence";
+const ROOM: &str = "room";
+const ROOM_INFO: &str = "room-info";
+const DISPLAY_NAME: &str = "display-name";
+const JOINED_USER_ID: &str = "joined-user-id";
+const INVITED_USER_ID: &str = "invited-user-ids";
```

Comment:
Assuming everything was changed to singular, you missed one case here.

```suggestion
const JOINED_USER_ID: &str = "joined-user-id";
const INVITED_USER_ID: &str = "invited-user-id";
```